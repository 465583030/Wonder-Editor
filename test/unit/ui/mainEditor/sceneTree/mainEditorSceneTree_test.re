open Wonder_jest;

open ExpectJs;

external toObject : ReactTestRenderer.t => Js.t({..}) = "%identity";

let _ =
  describe(
    "mainEditor sceneTree component",
    (_) => {
      open Expect;
      open Expect.Operators;
      open Sinon;
      let sandbox = getSandboxDefaultVal();
      beforeEach(
        () => {
          sandbox := createSandbox();
          TestToolEngine.prepare(sandbox)
        }
      );
      afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
      test(
        "create mainEditor sceneTree snapshot",
        (_) => {
          /* TestToolUI.initMainEditor(sandbox); */
          let component =
            ReactTestRenderer.create(
              <MainEditorSceneTree
                state=AppStore.state
                dispatch=(Reductive.Store.dispatch(IndexStore.store))
              />
            );
          let json = ReactTestRenderer.toJSON(component);
          toMatchSnapshot(expect(json))
        }
      )
    }
  );