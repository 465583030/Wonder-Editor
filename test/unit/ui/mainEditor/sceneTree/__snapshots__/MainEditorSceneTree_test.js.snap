// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorSceneTree ui component get scene tree from engine deal with the specific case test snapshot if drag treeNode into it's first layer chidlren, keep not change 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject7
        </li>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine deal with the specific case test snapshot if drag treeNode into it's second layer chidlren, keep not change 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject6
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "opacity": "1",
              }
            }
          >
            gameObject7
          </li>
        </ul>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine deal with the specific case test snapshot if drag treeNode into itself, keep not change 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject1
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject2
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine deal with the specific case test snapshot if drag treeNode move to canvas or other component, set css->opacity to 1 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject1
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject2
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test has children case have first layer children test snapshot drag first layer child treeNode into root div 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test has children case have first layer children test snapshot drag treeNode into first layer treeNode children 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject7
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "opacity": "1",
              }
            }
          >
            gameObject6
          </li>
        </ul>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test has children case have first layer children test snapshot drag treeNode into first layer treeNode parent 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject7
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject6
        </li>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test has children case have first layer children test snapshot no drag 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject4
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject7
        </li>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test has children case have second layer children drag has second treeNode into no child treNode 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject4
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "opacity": "1",
              }
            }
          >
            gameObject6
          </li>
          <ul
            className="wonder-tree-node"
            draggable={true}
            onDragEnd={[Function]}
            onDragStart={[Function]}
          >
            <li
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              gameObject7
            </li>
          </ul>
        </ul>
      </ul>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test simple scene graph data which haven't children case test snapshot drag treeNode into target treeNode 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
      <ul
        className="wonder-tree-node"
        draggable={true}
        onDragEnd={[Function]}
        onDragStart={[Function]}
      >
        <li
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "opacity": "1",
            }
          }
        >
          gameObject6
        </li>
      </ul>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`MainEditorSceneTree ui component get scene tree from engine test simple scene graph data which haven't children case test snapshot no drag 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;
